daemon/info_unix.go

```diff
@@ -58,28 +58,28 @@ func (daemon *Daemon) fillPlatformInfo(v *system.Info, sysInfo *sysinfo.SysInfo,
 	v.InitCommit.ID = "N/A"
 
 	if _, _, commit, err := parseDefaultRuntimeVersion(&cfg.Runtimes); err != nil {
-		log.G(context.TODO()).Warnf(err.Error())
+		log.G(ctx).Warnf(err.Error())
```

Comment:
We should return early if err was caused by context cancellation (and probably change this function to return an error?).

Reply:
⚠️ I haven't looked at the full context, so just a "blurb" to describe what I mean ❤️ ;

- Agreed on handling context cancellation
- For other _errors_ we must be "flexible";
    - the `info` endpoint is also used for debugging purposes; this means that some parts of it may _fail_ (e.g. unable to find containerd!)
    - so it should try to return "as much info as possible"
    - but not _fail_ to return anything if some pieces of information cannot be provided
    - ^^ perhaps we must better define "optional" and "required" information, but hope my general "train of thoughts" is clear here 😄

Reply:
Yeah, other errors are fine to continue, but if the error was caused by the context cancellation, there's just no point in doing anything more as the caller doesn't care about the result anymore.

Reply:
Yup! That's ok, and I agree with that.

Reply:
In this case there is no context passed into `parseDefaultRuntimeVersion` so no context cancellation.